Add UI for a setting to turn off double click to edit project file

davidwengier opened the pull request:
The code that checks this setting is in CPS.

Fixes #4493.

Note: This may not go ahead as the UI may be moved to the VS repo.

davkean commented on src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageResources.resx:
```diff
@@ -138,4 +138,7 @@
     <value>Use previews of the .NET Core SDK</value>
   </data>
+  <data name="General_DoubleClickOpenProjectFile" xml:space="preserve">
+    <value>Double click on project node to edit project file.</value>
```
This should not have a trailing ".".

davkean replied:
Also missing an access key.

davidwengier replied:
My WPF is weak... pointers?
![image](https://user-images.githubusercontent.com/754264/51980589-64fd8800-2488-11e9-9a12-8e1438ac86bc.png)
![image](https://user-images.githubusercontent.com/754264/51980572-5a42f300-2488-11e9-975a-db68aa814544.png)

drewnoakes replied:
Use a single underscore instead of an ampersand.

davidwengier replied:
WPF is weird. Bring back Windows Forms I say!

Thanks guys :P

drewnoakes replied:
`&` has a special meaning in XML. You'd have to use `&amp;`.

drewnoakes commented on src/Microsoft.VisualStudio.Editors/OptionPages/GeneralOptionPageControl.xaml:
```diff
@@ -17,6 +17,9 @@
             </StackPanel>
+            <StackPanel Orientation="Horizontal">
+                <CheckBox x:Name="DoubleClickOpenProjectFile" x:Uid="General_DoubleClickOpenProjectFile" Content="{x:Static r:GeneralOptionPageResources.General_DoubleClickOpenProjectFile}" Margin="13, 10" />
+            </StackPanel>
```
Stack panels with a single child aren't necessary.

davkean commented:
Given https://dev.azure.com/devdiv/DevDiv/_workitems/edit/779596?src=WorkItemMention&src-action=artifact_link, and the more I think about this, I'm thinking we might want to push this setting along with the Preview settings instead of shipping the org chart here.

davidwengier commented:
Hadn't seen that. Makes the UI tricky, as a setting for previewing .NET SDK project files would apply to all three of the current preview window sub-settings, and I would have thought we'd still want a separate setting for the double click behaviour.

![image](https://user-images.githubusercontent.com/754264/52033783-100c5100-251e-11e9-8ac3-a6c570b22f27.png)

davkean commented:
I'd say that it's only relevant for Solution Explorer, though I'd need to ask the customer what exactly bothers them about it.

If you wanted to switch both, it's weird that one would be under .NET Core and one under that.

davidwengier commented:
For sure the Tabs and Windows section is a better spot for both settings, just not sure what layout would make sense.

davidwengier commented:
Renamed the setting based on feedback from the UX meeting today, and changed a couple of other strings even though they're not used, because they might be once #4561 is done.

davidwengier commented:
UX board has recommended against adding an option to control this behaviour.
